tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1514,24 +1519,193 @@ internal string GetStampPropertyValue(Dictionary<string, object> stampedProperty
+
+                jw.WritePropertyName("Fields");
+                jw.WriteStartArray();
+                foreach (var field in fieldCollection)
+                {
+                    if (field.Group == this.contentTypesSettings.OneDriveContentTypeGroup)
+                    {
+                        jw.WriteStartObject();
+                        jw.WritePropertyName("name");
+                        jw.WriteValue(field.Title);
+
+                        jw.WritePropertyName("fieldInternalName");
+                        jw.WriteValue(field.InternalName);
+
+                        jw.WritePropertyName("required");
+                        string isRequired = AddFields.Count > 0 ? AddFields.Where(x => x.FieldName == field.InternalName).SingleOrDefault().IsMandatory : field.Required.ToString();
+                        field.Required = string.IsNullOrWhiteSpace(isRequired) ? false : Convert.ToBoolean(isRequired);
+                        jw.WriteValue(field.Required);
+
+                        jw.WritePropertyName("displayInUI");
+                        string isDisplayInUI = AddFields.Count > 0 ? AddFields.Where(x => x.FieldName == field.InternalName).SingleOrDefault().IsDisplayInUI : "true";
+                        isDisplayInUI = string.IsNullOrWhiteSpace(isDisplayInUI) ? "false" : isDisplayInUI;
+                        jw.WriteValue(isDisplayInUI);
+
+                        jw.WritePropertyName("originalType");
+                        jw.WriteValue(field.TypeAsString);
+                        jw.WritePropertyName("defaultValue");
+                        jw.WriteValue(field.DefaultValue);
+                        jw.WritePropertyName("description");
+                        jw.WriteValue(field.Description);
+
+                        if (field.TypeAsString == "Choice")
+                        {
+                            jw.WritePropertyName("type");
+                            jw.WriteValue(Convert.ToString(((Microsoft.SharePoint.Client.FieldChoice)field).EditFormat));
+                            List<string> options = GetChoiceFieldValues(clientContext, field);
+                            jw.WritePropertyName("values");
+                            jw.WriteStartArray();
+                            int optionCounter = 1;
+
+                            foreach (string option in options)
+                            {
+                                jw.WriteStartObject();
+                                jw.WritePropertyName("choiceId");
+                                jw.WriteValue(optionCounter);
+                                jw.WritePropertyName("choiceValue");
+                                jw.WriteValue(option);
+                                optionCounter++;
+                                jw.WriteEndObject();
+                            }
+                            jw.WriteEndArray();
+                        }
+                        else if (field.TypeAsString == "MultiChoice")

```

Comment:
Logic to create json object for multichoice type field. Currently we are supporting Text, RadioButton, Dropdown, MultipleChoice, dateTime , Checkbox type field.